src/vuex/modules/index/store.ts

```diff
@@ -104,33 +101,55 @@ const actions = createActions({
     removeIndex(index)
     commit.deleteIndex(index)
   },
-  async listIndexesAndCollections(context) {
+  async listIndexes(context) {
     const { commit } = indexActionContext(context)
-    commit.setLoading(true)
-    commit.reset()
+    commit.setLoadingIndexes(true)
     let result = await Vue.prototype.$kuzzle.index.list()
-
-    let indexesAndCollections = {}
     result = result.filter(index => index !== '%kuzzle')
+
     for (const index of result) {
-      const res = await Vue.prototype.$kuzzle.collection.list(index, {
-        size: 0
-      })
+      commit.addIndex(index)
+    }
+    commit.setLoadingIndexes(false)
+  },
+  async listCollectionsForIndex(context, index) {
+    const { commit } = indexActionContext(context)
+    commit.setLoadingCollectionsForIndex({ index, loading: true })
+    const res = await Vue.prototype.$kuzzle.collection.list(index, {
+      size: 0
+    })
+    // debugger
+    let collections = splitRealtimeStoredCollections(res.collections)
 
-      let collections = splitRealtimeStoredCollections(res.collections)
+    if (!collections.realtime) {
+      collections.realtime = []
+    }
+
+    collections.realtime = collections.realtime.concat(
+      getRealtimeCollectionFromStorage(index)
+    )
+    collections = dedupeRealtimeCollections(collections)
+    commit.setCollectionsForIndex({
+      index,
+      collections: collections.stored,
+      type: 'stored'
+    })
+    commit.setCollectionsForIndex({
+      index,
+      collections: collections.realtime,
+      type: 'realtime'
+    })
+    commit.setLoadingCollectionsForIndex({ index, loading: false })
+  },
 
-      if (!collections.realtime) {
-        collections.realtime = []
-      }
+  async listIndexesAndCollections(context) {
+    const { commit, dispatch } = indexActionContext(context)
+    commit.reset()
 
-      collections.realtime = collections.realtime.concat(
-        getRealtimeCollectionFromStorage(index)
-      )
-      collections = dedupeRealtimeCollections(collections)
-      indexesAndCollections[index] = collections
-      commit.receiveIndexesCollections(indexesAndCollections || {})
-    }
-    commit.setLoading(false)
+    await dispatch.listIndexes()
+    Object.keys(state.indexesAndCollections).forEach(async index => {
+      await dispatch.listCollectionsForIndex(index)
```

Comment:
Awaiting a promise inside a forEach won't really wait for the promise to complete before processing the next item on the loop.  
This method will return before each promise is fulfilled, I don' know if this is the intended behavior.

Reply:
Thanks! I forgot to delete the `await`